ponycheck/property.pony

```diff
@@ -143,3 +89,103 @@ primitive Stringifier
         "<identity:" + digest.string() + ">"
       end
     (consume t, consume s)
+
+class iso Property1UnitTest[T] is UnitTest
+  let _prop1: Property1[T]
+
+  new iso create(p1: Property1[T]) =>
+    _prop1 = p1
+
+  fun name(): String =>
+    _prop1.name()
+
+  fun ref apply(h: TestHelper) =>
+    h.long_test(_prop1.params().timeout)
+    _Runner[T](_prop1, h).run()
+
+actor _Runner[T]
+  let _prop1: Property1[T]
+  let _h: TestHelper
+  let _rnd: Randomness
+  let _ph: PropertyHelper
+  var _shrinker: Iterator[T^] = _EmptyIterator[T^]
+
+  new create(p1: Property1[T], h: TestHelper) =>
+    _prop1 = consume p1
+    _h = consume h
+    _rnd = Randomness(_prop1.params().seed)
+    _ph = PropertyHelper(_prop1.params(), _h)
+
+  be run(n: USize = 0) =>
+    if n == _prop1.params().num_samples then
+      complete()
+      return
+    end
+    (var sample, _shrinker) = _prop1.gen().generate_and_shrink(_rnd)
```

Comment:
the `gen()` method should only be called once to create the constructor. I'd suggest doing it in the constructor and assign it to a field.